ferrocene/doc/internal-procedures/src/testing-other-targets.rst

```diff
@@ -0,0 +1,193 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You need to use WSL2. It's recommended to use the latest stable Ubuntu.
+
+Here is a sample configuration for your ``/etc/wsl.conf`` (in the guest):
+
+.. code-block::
+
+    [user]
+    default=ana
+
+    [boot]
+    systemd=true
+
+    [wsl2]
+    nestedVirtualization=true
+    
+If you changed your configuration, make sure to restart the environment with ``wsl --shutdown``.
+
+Enter the shell with ``wsl``, or `point Visual Studio Code <https://code.visualstudio.com/docs/remote/wsl-tutorial>`_ at it.
+
+Target Procedures
+-----------------
+
+In general, bare-metal targets follow a similar strategy: Use `binfmt_misc` to run the
+test target binaries.
+
+:ref:`aarch64-unknown-none`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. Note::
+    
+    In a ``aarch64-unknown-linux-gnu`` environment you may simply skip to the final step,
+    no QEMU is needed.
+
+Install the necessary packages:
+
+.. code-block:: bash
+
+    sudo apt install gcc-aarch64-linux-gnu qemu-system-aarch64
+
+If you don't already have a ``/usr/share/binfmts/qemu-aarch64`` file, create one:
+
+.. code-block:: bash
+
+    package qemu-aarch64
+    interpreter /usr/bin/qemu-aarch64-static
+    magic \x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\xb7\x00
+    mask \xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff
+    credentials no
+    preserve no
+    fix_binary no
+    
+Then make sure it's imported:
+
+.. code-block:: bash
+    
+   sudo update-binfmts --import qemu-aarch64
+
+You can now run the tests:
+
+.. code-block:: bash
+
+    ./x test --stage 1 --target aarch64-unknown-ferrocenecoretest library/core
+
+:target:`thumbv7em-none-eabihf` & :target:`thumbv7em-none-eabi`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Install the necessary packages:
+
+.. code-block:: bash
+
+    sudo apt install gcc-arm-none-eabi qemu-system-arm
+
+If you don't already have a ``/usr/share/binfmts/qemu-arm`` file, create one:
+
+.. code-block:: bash
+
+    package qemu-arm
+    interpreter /usr/bin/qemu-arm-static
+    magic \x7fELF\x01\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x28\x00
+    mask \xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff
+    credentials no
+    preserve no
+    fix_binary no
+    
+Then make sure it's imported:
+
+.. code-block:: bash
+    
+   sudo update-binfmts --import qemu-arm
+
+Currently, this target uses our *secret sauce*. Eventually this will be an open source component of Ferrocene, but for now, it's our little bit of arcane magic.
+
+First, set the target:
+
+.. code-block:: bash
+
+    export TARGET="thumbv7em-ferrocenecoretest-eabihf"
+    # or 
+    export TARGET="thumbv7em-ferrocenecoretest-eabi"
+
+Next, it's possible to build the *secret sauce*, or to download it. Generally, it's easier to download it, but if necessary you can find the repository in the `Ferrocene <https://github.com/ferrocene>`_ organization.
```

Comment:
How about 

```suggestion
Download the secret sauce FROM WHERE?, or you can build it from the repository in the `Ferrocene <https://github.com/ferrocene>`_ organization if necessary.
```
should we point to which repo?

Reply:
I'd prefer to keep our cards close on this one. The repository has setup instructions, and it is deliberately secret for the time being. Any of our internal developers can easily find it using the search for "sauce" or "secret". We'll eventually open source it and these steps will vanish.

Reply:
TBH we shouldn't manually download stuff and put it in a known location in `/tmp`, also because it has the risk of weird/unintuitive failures if somehow you use an old cached version. But this is not a documentation problem, rather a tooling problem. Let's followup this in Zulip.

Reply:
Ya -- agree, would be very happy to improve the state of this, but this document describes it as-it-exists.